lib/trino-parquet/src/main/java/io/trino/parquet/reader/FilteredOffsetIndex.java

```diff
@@ -112,29 +112,22 @@ public String toString()
                 ranges.add(new OffsetRange(rowGroupOffset, firstPageOffset - rowGroupOffset));
             }
 
-            long currentOffset = -1;
-            long currentLength = 0;
-            for (int i = 0; i < pageCount; ++i) {
+            long currentOffset = getOffset(0);
+            long currentLength = getCompressedPageSize(0);
+            for (int i = 1; i < pageCount; ++i) {
                 long offset = getOffset(i);
                 int length = getCompressedPageSize(i);
 
-                if (currentOffset == -1) {
-                    currentOffset = offset;
-                    currentLength = length;
-                }
-                else if (currentOffset + currentLength == offset) {
+                if (currentOffset + currentLength == offset) {
                     currentLength += length;
                 }
                 else {
                     ranges.add(new OffsetRange(currentOffset, currentLength));
-                    currentOffset = -1;
-                    currentLength = 0;
+                    currentOffset = offset;
+                    currentLength = length;
```

Comment:
Looks like this would be the minimal fix, right?

I like other changes, but would be nice to separate actual bugfix from readability improvement.

Reply:
Yes, I confirmed by running tests that this would fix it too.
If you don't mind, I'd prefer to keep the change in current commit though as I found it easier to follow the logic after simplification of the loop.

Reply:
Yes, i find your code more readable. However, i think it would be better to

- have minimal change fixing the problem (those two lines), along with test you added
- have next commit do the readability refactor

this way, someone else looking at the fix commit, will have it easy to understand what was the problem about (we were losing offset,length pairs)

Reply:
I've separated the simplification into another commit now